test/e2e/initialize.sh

```diff
@@ -84,11 +85,11 @@ done
 if [ "$all_exist" = false ] || [ "$current_hash" != "$stored_hash" ]; then
     echo "WASM files are missing or contracts have been updated. Initializing and building contracts..."
     # Initialize contracts
-    $soroban contract init "$dirname/test-contracts" --with-example increment other_custom_types atomic_swap token
-    
+    npx degit --force AhaLabs/soroban-test-examples "$dirname/test-contracts"
```

Comment:
This is using a new "test-examples" repo at https://github.com/AhaLabs/soroban-test-examples. These are test-specific examples, rather than being learner-facing examples like the main examples repo. For more motivations on this, see https://github.com/stellar/soroban-examples/pull/322#issuecomment-2272560270 

This new repo should probably live in the stellar org. Happy to move it, if you'd like!

Reply:
We're already using `git` above; why can't we just clone it? Submodules are fine too but I don't really care since this isn't user-facing -- whatever is easier for y'all to maintain works for me :+1:

Reply:
Cloning is error-prone (can fail your CI randomly during testing and it's getting cloned every time), submodules are not (you clone the repo for testing once and then it's cached)

If we go with git clone approach, we should pin the commit hash as well and clone without history

Reply:
Ah, good point re: transient failures. I'm good with a submodule, then :+1:

Reply:
Added as git submodule! Still using the AhaLabs repo; should we wait until it moves to merge this PR?

Reply:
Updated!